latch/ldata/path.py

```diff
@@ -0,0 +1,272 @@
+            query GetNodeData($path: String!) {
+                ldataResolvePathToNode(path: $path) {
+                    path
+                    ldataNode {
+                        finalLinkTarget {
+                            id
+                            name
+                            type
+                            removed
+                            ldataObjectMeta {
+                                contentSize
+                                contentType
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"path": self.path},
+        )["ldataResolvePathToNode"]
+
+        if data is None or data["ldataNode"] is None:
+            raise FileNotFoundError(f"No such Latch file or directory: {self.path}")
+
+        self._cache.path = self.path
+
+        final_link_target = data["ldataNode"]["finalLinkTarget"]
+        self._cache.node_id = final_link_target["id"]
+        self._cache.name = final_link_target["name"]
+        self._cache.type = LDataNodeType(final_link_target["type"].lower())
+
+        meta = final_link_target["ldataObjectMeta"]
+        if meta is not None:
+            self._cache.size = (
+                -1
+                if meta["contentSize"] is None
+                else int(final_link_target["ldataObjectMeta"]["contentSize"])
+            )
+            self._cache.content_type = final_link_target["ldataObjectMeta"][
+                "contentType"
+            ]
+
+    def node_id(self, *, load_if_missing: bool = True) -> str:
+        match = node_id_regex.match(self.path)
+        if match:
+            return match.group("id")
+
+        if self._cache.node_id is None or self._cache.path != self.path:
+            if not load_if_missing:
+                return None
+            self.load()
+        return self._cache.node_id
+
+    def name(self, *, load_if_missing: bool = True) -> str:
+        if self._cache.name is None or self._cache.path != self.path:
+            if not load_if_missing:
+                return None
+            self.load()
+        return self._cache.name
+
+    def type(self, *, load_if_missing: bool = True) -> LDataNodeType:
+        if self._cache.type is None or self._cache.path != self.path:
+            if not load_if_missing:
+                return None
+            self.load()
+        return self._cache.type
+
+    def size(self, *, load_if_missing: bool = True) -> float:
+        if self._cache.size is None or self._cache.path != self.path:
+            if not load_if_missing:
+                return None
+            self.load()
+        return self._cache.size
+
+    def content_type(self, *, load_if_missing: bool = True) -> str:
+        if self._cache.content_type is None or self._cache.path != self.path:
+            if not load_if_missing:
+                return None
+            self.load()
+        return self._cache.content_type
+
+    def is_dir(self) -> bool:
```

Comment:
```suggestion
    def is_dir(self, *, load_if_missing) -> bool:
```
and pass it down to the `type` call